pkg/cloudprovider/gcp/block_store.go

```diff
@@ -111,19 +127,44 @@ func (b *blockStore) CreateVolumeFromSnapshot(snapshotID, volumeType, volumeAZ s
 		Description:    res.Description,
 	}
 
-	if _, err = b.gce.Disks.Insert(b.project, volumeAZ, disk).Do(); err != nil {
-		return "", errors.WithStack(err)
+	if isMultiZone(volumeAZ) {
+		volumeRegion, err := parseRegion(volumeAZ)
+		if err != nil {
+			return "", errors.WithStack(err)
```

Comment:
you don't need to wrap the error using `errors.WithStack` once you switch L106 to `errors.Errorf` since that will give you a stack trace at the call site